### ISSUE TITLE
:question: Move away from enforcing the use of a protocol?

### ISSUE THREAD

**Issue by tombulled** [neoclient/neoclient#68]
Using a protocol (interface) is awesome and should be encouraged, however should it be enforced?

Perhaps a `Controller` can be used (which ultimately may implement a protocol).

For example:
```python
from neoclient import Controller, NeoClient, get

client: NeoClient = NeoClient("https://httpbin.org/")

@client.controller
class Httpbin(Controller):
    @get("/ip")
    def ip(self):
        ...
httpbin: Httpbin = Httpbin()
```

or, to use an interface:
```python
from typing import Protocol
from neoclient import Controller, NeoClient, get

class Httpbin(Protocol):
    def ip(self):
        ...

class HttpbinController(Controller):
    @get("/ip")
    def ip(self):
        ...

client: Httpbin = HttpbinController()
```

**Comment by tombulled** [comment]
Could also implement a "resource":

```python
from neoclient import resource

@resource("/user")
class UserResource:
    @post
    def create(self, user: User):
        ...

    @get
    def get(self, id: str):
        ...
```

**Comment by tombulled** [comment]
Resolved under #80
